osquery/tables/applications/linux/docker_deb_packages.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/applications/linux/namespace_ops.h>
+#include <osquery/tables/applications/posix/docker_api.h>
+#include <osquery/tables/system/linux/deb.h>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include <iostream>
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include "osquery/core/json.h"
+#include "rapidjson/filereadstream.h"
+
+namespace osquery {
+namespace tables {
+
+void extractDebPackageInfo(const struct pkginfo* pkg,
+                           rapidjson::Writer<rapidjson::StringBuffer>& w) {
+  struct varbuf vb;
+  varbuf_init(&vb, 20);
+  w.StartObject();
+
+  // Iterate over the desired fieldinfos, calling their fwritefunctions
+  // to extract the package's information.
+  const struct fieldinfo* fip = nullptr;
+  for (fip = fieldinfos; fip->name; fip++) {
+    fip->wcall(&vb, pkg, &pkg->installed, fw_printheader, fip);
+
+    std::string line = vb.string();
+    if (!line.empty()) {
+      size_t separator_position = line.find(':');
```

Comment:
It may be worth being a little more explicit about the format expected. Can a non-empty line validly have no separator? Can it have multiple ones? If a non-empty line without a separator is invalid, that should be caught.

Reply:
Also separator_position maybe == `std::string::npos`. Should probably test for that